pkg/migration/controllers/migration.go

```diff
@@ -676,6 +676,66 @@ func (m *MigrationController) updateResourceStatus(
 	}
 }
 
+func (m *MigrationController) getRemoteAdminConfig(migration *stork_api.Migration) (*kubernetes.Clientset, error) {
+	remoteConfig, err := getClusterPairSchedulerConfig(migration.Spec.ClusterPair, migration.Namespace)
+	if err != nil {
+		return nil, err
+	}
+	remoteAdminConfig := remoteConfig
+	// Use the admin cluter pair for cluster scoped resources if it has been configured
+	if migration.Spec.AdminClusterPair != "" {
+		remoteAdminConfig, err = getClusterPairSchedulerConfig(migration.Spec.AdminClusterPair, m.migrationAdminNamespace)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	adminClient, err := kubernetes.NewForConfig(remoteAdminConfig)
+	if err != nil {
+		return nil, err
+	}
+	return adminClient, nil
+}
+
+func (m *MigrationController) checkAndUpdateDefaultSA(
+	migration *stork_api.Migration,
+	object runtime.Unstructured,
+) error {
+	var sourceSA v1.ServiceAccount
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.UnstructuredContent(), &sourceSA); err != nil {
+		return fmt.Errorf("error converting to serviceAccount: %v", err)
+	}
+	adminClient, err := m.getRemoteAdminConfig(migration)
+	if err != nil {
+		return err
+	}
+
+	if sourceSA.GetName() != "default" {
+		// delete and recreate service account
+		if err := adminClient.CoreV1().ServiceAccounts(sourceSA.GetNamespace()).Delete(sourceSA.GetName(), &metav1.DeleteOptions{}); err != nil {
+			return err
+		}
+		if _, err := adminClient.CoreV1().ServiceAccounts(sourceSA.GetNamespace()).Create(&sourceSA); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	log.MigrationLog(migration).Infof("Updating default service account with image pull secrets")
+	// merge service account resource for default namespaces
+	destSA, err := adminClient.CoreV1().ServiceAccounts(sourceSA.GetNamespace()).Get(sourceSA.GetName(), metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	destSA.ImagePullSecrets = append(destSA.ImagePullSecrets, sourceSA.ImagePullSecrets...)
```

Comment:
Can you confirm that this works fine even if the destination service account already has imagePullSecrets with the same name.

Reply:
it works even though destination service account already has imagePullSecrets, it just add duplicate entry under image pull secrets 
eg. 
```
Name:                default
Namespace:           default
Labels:              <none>
Annotations:         <none>
Image pull secrets:  test-dr
                     test-dr
Mountable secrets:   default-token-44krf
Tokens:              default-token-44krf
Events:              <none>
```

So I added commit to not duplicated already present `imagepullsecrets`